beacon-chain/db/kv/blocks.go

```diff
@@ -80,6 +80,21 @@ func (s *Store) OriginCheckpointBlockRoot(ctx context.Context) ([32]byte, error)
 	return root, err
 }
 
+// HeadBlockRoot returns the latest canonical block root in the Ethereum Beacon Chain.
+func (s *Store) HeadBlockRoot() ([32]byte, error) {
+	var root [32]byte
+	err := s.db.View(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(blocksBucket)
+		headRoot := bkt.Get(headBlockRootKey)
+		if headRoot == nil {
```

Comment:
nitpick: I looked at the `Get` code to make sure there's no way for a "not found" value to be an empty slice, so I'm confident what you have here works, but as a general rule, I think we should prefer `len(headRoot) == 0` over `headRoot == nil` when checking byte slices.